### PR TITLE
feat(cli): deploy command should allow customizing git config user.name / user.email in deployment branch (#9101)

### LINKED ISSUE #9101: Enhance the deploy command to enable the setting of git user.name and user.email in deployment branch
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Description

Enhance the deploy command to enable the setting of git user.name and user.email specifically for the deployment branch. This way, Git will utilize the local git configuration instead of the global configuration when making commits to the deployment branch.

### Has this been requested on Canny?

NA

### Motivation

In my system, I have separate configurations for my personal and work Git users. These configurations are set on a per-repository basis rather than globally. However, when I use the command "yarn deploy," it automatically sets the user.name and user.email to default values, which are based on my hostname. I would rather not share this information on GitHub and have the option to set the user.name and user.email that I would like to use for commiting changes to deployment branch.

### API design

I have added 2 environment variables to set the user.name and user.email  

### Have you tried building it?

Yes, I have added 2 environment variables to set the user.name and user.email  

### Self-service

- [X] I'd be willing to contribute this feature to Docusaurus myself.

### PR DESCRIPTION
**Pull request by amdshrif**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

Fix #9101

### Test links

Deploy preview: https://deploy-preview-_____--docusaurus-2.netlify.app/

### CODE REVIEW

File: packages/docusaurus/src/commands/deploy.ts [hunk 1]
```diff
@@ -215,6 +215,16 @@ You can also set the deploymentBranch property in docusaurus.config.js .`);
...
 
+    const gitUserName = process.env.GIT_USER_NAME;
+    if (gitUserName) {
+      shellExecLog(`git config user.name "${gitUserName}"`);
+    }
+
+    const gitUserEmail = process.env.GIT_USER_EMAIL;
+    if (gitUserEmail) {
+      shellExecLog(`git config user.email "${gitUserEmail}"`);
+    }
```
Comment (slorber) [inline packages/docusaurus/src/commands/deploy.ts:226]: not super fan of having config side-effects like this, and features without documentation

**Review by slorber — CHANGES_REQUESTED** [review]
Do we really need this?

What prevents from replacing:

```bash
export GIT_USER_NAME="xyz"
export GIT_USER_EMAIL="my@email.com"
yarn deploy
```

With:

```bash
git config user.name "xyz"
git config user.email "my@email.com"
yarn deploy
```

**Comment by amdshrif** [comment]
> Do we really need this?
> 
> What prevents from replacing:
> 
> ```shell
> export GIT_USER_NAME="xyz"
> export GIT_USER_EMAIL="my@email.com"
> yarn deploy
> ```
> 
> With:
> 
> ```shell
> git config user.name "xyz"
> git config user.email "my@email.com"
> yarn deploy
> ```

The suggested enhancement will only update the user.name and user.email locally on the repo level, while keeping the global git config as-is and I do not have to bother to keep switching between the emails and names for every work I do on my machine. 

The new enhancement works well with a local shell script that exports the other env variable settings all together. 

env-setting.sh
```shell 
export USE_SSH=true
export GIT_USER="amdshrif"
export CURRENT_BRANCH="deployment"
export GIT_USER_NAME="Ahmad Shrif"
export GIT_USER_EMAIL="ahmad.shrif@gmail.com"
```
source env-settings.sh

**Comment by slorber** [comment]
> The suggested enhancement will only update the user.name and user.email locally on the repo level, while keeping the global git config as-is and I do not have to bother to keep switching between personal and work emails for every work I do on my machine.

Sorry, but I don't understand your answer.

> The suggested enhancement will only update the user.name and user.email locally on the repo level, while keeping the global git config as-is

Isn't it the intended behavior in both cases?

---

I'll ask again:

Does the following work to deploy with the specified username/email, yes or not?

```sh
git config user.name "Ahmad Shrif"
git config user.email "ahmad.shrif@gmail.com"
yarn deploy
```

Have you tested it? 

If it does not work, can you tell me what's the behavior you see, and what's the behavior you want?

**Comment by slorber** [comment]
Ok I understand the problem now, there are 2 repositories involved:
- The current one
- The one we clone for deployment (possibly a different one)

We need to set the config on the cloned deployment

I'm ok to merge this, but please add some documentation for these env variables

**Comment by amdshrif** [comment]
> Ok I understand the problem now, there are 2 repositories involved:
> 
> * The current one
> * The one we clone for deployment (possibly a different one)
> 
> We need to set the config on the cloned deployment
> 
> I'm ok to merge this, but please add some documentation for these env variables

Yes, that the is correct. Actually it is one repository and two branches, when we run `yarn deploy` it does a clone for the repository with the selected branch in a tmp folder and pushes the changes from there.

I will update the PR to update the docs and testing.

**Comment by slorber** [comment]
Thanks, added the docs to your pr already :)
